frontend/src/pages/SignUpPage.tsx

```diff
@@ -1,46 +1,41 @@
-import { useCallback, FormEvent } from "react";
+import { useCallback } from "react";
 
 import { Container } from "~/shared/components/Container";
 import { UserApi } from "~/generated";
+import { SignForm } from "~/shared/components/SignForm";
 
 export const SignUpPage = () => {
-  const handleSubmit = useCallback(async (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    const form = event.currentTarget;
-    const formData = new FormData(form);
-    const username = formData.get("username") as string;
-    const password = formData.get("password") as string;
-
+  const handleSubmit = useCallback(async (username: string, password: string) => {
     const userApi = new UserApi();
-    const response = await userApi.signUp({
-      username,
-      password,
-    }, {
-      withCredentials: true,
-    });
-
-    if (response.status === 200) {
-      console.log("Success");
-
-      window.location.href = "/sign-in";
+    
+    try {
+      const response = await userApi.signUp({
+        username,
+        password,
+      }, {
+        withCredentials: true,
+      });
+  
+      if (response.status === 200) {
+        window.location.href = "/sign-in";
+        
+        return;
+      }
+    } catch (error) {
+      alert("サインアップに失敗しました。ユーザーネームが既に使用されている可能性があります");
```

Comment:
[nits]
利用者視点だとユーザー名の方が親切かも
```suggestion
      alert("サインアップに失敗しました。ユーザー名が既に使用されている可能性があります");
```